### DISCUSSION TITLE
Feature request : outline editor

### DISCUSSION THREAD

**Discussion by antoineschmitt** [standardnotes/forum#2483] [Suggestions]
More than a task editor, it would be nice to have an outline editor.

Outline editors are great for managing nested texts. There are many outline editors, and many notes editors, and not one does boths, not even Evernote. This feature, alongside all of the great features of StandardNotes (sync, multiplatform, encryption) would make it a killer app.

**Comment by atmoio** [comment]
Would indeed be cool. If you know of any open-source outliners, that would be helpful. Would be difficult to develop one from scratch.

**Comment by greiggy** [comment]
Would give SN a significant feature advantage over other comparable notes apps! Judging by the enduring popularity of outliners, and new outliner apps appearing, I think this would be attractive to many writers.

**Comment by atmoio** [comment]
What would be, in your opinion, the most simple acceptable form of an outline editor? What is the bare minimum features it must have?

**Comment by greiggy** [comment]
A bit of structure, H1-H3, text, sub-text notes. With indenting of the
levels.

A more adventurous approach and much more functional, would have
expand/collapse of headings. That might be a bit more than an afternoon's
work...:8-)

Thanks for listening, IanG

**Comment by ajvsol** [comment]
Ideally this would be something like [WorkFlowy](https://workflowy.com/demo/embed/) with it's core features of sub-text notes, as well as collapsible and zoomable bullet points. [Dynalist](http://dynalist.io/) extends this with a lot more features but the basics in WF would be sufficient for my usecase.

**Comment by ajvsol** [comment]
Also [Calculist](https://github.com/calculist/calculist) is an open-source outliner with similar functionality.

**Comment by atmoio** [comment]
This one looks pretty cool: https://github.com/WuTheFWasThat/vimflowy

**Reply by FractalHabits** [reply]
This would be great. RemNote has a great one like this, where you can collapse any bullet and any bullet can also be a task.

**Comment by greiggy** [comment]
Yes, looks cool and seems to have all the right actions.

--IanG

Ian Greig

**Comment by YigitDemirag** [comment]
With this feature, we can also properly import OPML files.

**Comment by chillgoze** [comment]
Would love this feature too, specially if it could allow entering and indenting text with minimum keystrokes (e.g. Dynalist, Checkvist). The next level of maturity on this feature would be able to extend it to mind-maps :).

**Comment by michalradacz** [comment]
I find some outline editor extension on github. But I am not familiar with NPN tool to build it on my windows machine, so I am not able to import this extension :-(

**Comment by JaspalSuri** [comment]
Do you have a link to the repo?

**Comment by michalradacz** [comment]
https://github.com/sn-extensions/outline-editor



Odesláno z iPhonu

16. 6. 2020 v 4:09, Jaspal Suri <notifications@github.com>:

﻿

Do you have a link to the repo?

**Comment by JaspalSuri** [comment]
Oh, it's one of the Extensions under our repo! 

It seems like it has been put on hold and hasn't been updated in a year, so it would have to be something that we'd revisit in the future.

**Comment by michalradacz** [comment]
Thanks for info. But, it will be nice if you want to continue to dev first version, because outliner for SN is great function. I cannot use Checkvist because its not accessible, I am using SN and using for outlines will be great.


Odesláno z aplikace Pošta<https://go.microsoft.com/fwlink/?LinkId=550986> pro Windows 10

Od: Jaspal Suri<mailto:notifications@github.com>
Odesláno:středa 17. června 2020 8:14
Komu: standardnotes/forum<mailto:forum@noreply.github.com>
Kopie: Michal Rada<mailto:michal@kancelarmr.cz>; Comment<mailto:comment@noreply.github.com>
Předmět: Re: [standardnotes/forum] Feature request : outline editor (#180)


Oh, it's one of the Extensions under our repo!

It seems like it has been put on hold and hasn't been updated in a year, so it would have to be something that we'd revisit in the future.

**Comment by harvest316** [comment]
@TheodoreChu Can we please bump this request up again please?  I'm trying to transition from Workflowy and this is a blocking feature for me.

**Comment by myreli** [comment]
There are some extensions published here https://github.com/jonhadfield/awesome-standard-notes#editors that could help.

**Comment by mwcarlsn** [comment]
Looks like this has been an open request since 2018.     Having multi-level outliner functionality with expand/collapse features would be so incredibly helpful.

I have been looking around the interweb for the last year and there are zero apps that can outline  AND are highly-secure.

I have been using CheckVist since it started in 2012, but I can't trust it with sensitive data.    For sensitive data, I have using secure notes in LastPass and TrueKey and BitWarden, but the first two have security and support issues and BitWarden is unable to get full text search working on their android app.  Also, NONE of these 'security' apps have any Outlining capabilities.

Standard Notes has been my Go-To for all my text notes, but for outlining I just can't get away from the power of CheckVist.   PLEASE bring that power into Standard Notes so I can be sure my outline notes are just as secure as my text notes.

Thanks!!
-mc

**Reply by affvalente** [reply]
Agree this would be very good to add

**Comment by FractalHabits** [comment]
This would be great. RemNote has a great one but isn't secure. It allows you to collapse any bullet and any bullet can also be a task. This is a super important feature to be able  to hold extra info about a parent without too much clutter.
